[Docs]: Improve explanation about recycle bin (#15633)
pimcore/pimcore: doc/Development_Documentation/22_Administration_of_Pimcore/02_Cleanup_Data_Storage.md

mysql -e "TRUNCATE TABLE ###.recyclebin;"
rm -r var/recyclebin
```

> **WARNING: The recycle bin is an administrative tool and displays all the deleted elements by any user. 

Reviewer: ```suggestion
**WARNING: The recycle bin is an administrative tool that displays any user's deleted elements. 
```

> Due to the nature and complexity of element deletion and restoration process, this should be reserved to administrator and advanced users**

Reviewer: ```suggestion
Due to the nature and complexity of the elements deletion and restoration process, this tool should be reserved for administrator and advanced users**
```
